### DISCUSSION TITLE
g++-14 quick_exit and -D_POSIX_C_SOURCE

### DISCUSSION THREAD

**Discussion by marekkokot** [Homebrew/discussions#5714] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 91f3671b16fe62121364c465c14e0dfd088d7ead
Last commit: 9 days ago
Core tap JSON: 01 Nov 12:25 UTC
Core cask tap JSON: 01 Nov 12:25 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BROWSER: /Users/mkokot/.vscode-server/cli/servers/Stable-65edc4939843c90c34d61f4ce11704f09d3e5cb6/server/bin/helpers/browser.sh
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hello,

We use a igraph that is build using CMake for which we force using homebrew g++ on mac os.
It does not compile with g++-14 from homebrew. I narrowed it down, and the problem is there if `-D_POSIX_C_SOURCE=200809L` is set and if `cstdlib` is included.
For example, in this code (`test.cpp`):
```
#include <cstdlib>
 
int main(){
}
```

If I compile
`g++-14 test.cpp`
it compiles, but if I use
`g++-14  -D_POSIX_C_SOURCE=200809L test.cpp`
I get:
![image](https://github.com/user-attachments/assets/489eed43-5253-4c68-acd2-69ed45c02ec0)

This compiles just fine with g++-13, g++-12, and g++-11, and on linux machine with g++-14 this also compiles as far as I remember.

I'm not sure if this is the right place to post this. If not, please let me know where to post this issue (I am not sure if others can reproduce it).

**Comment by szhorvat** [comment]
Maybe this should be an issue instead of a discussion.
